core/src/client/ErrorHandler/ErrorHandler.tsx

```diff
@@ -18,38 +18,51 @@ interface ErrorHandlerState {
     showDetails: boolean;
 }
 
-export class ErrorHandler extends React.PureComponent<ErrorHandlerProps, ErrorHandlerState> {
-    constructor(props) {
-        super(props);
+export class ErrorHandler extends PureComponent<ErrorHandlerProps, ErrorHandlerState> {
 
-        this.state = {
-            showDetails: false,
-        };
-    }
+    state: Readonly<ErrorHandlerState> = { showDetails: false };
 
     onBackClick = (): void => {
-        window.history.back();
+        // Back button - takes you back to the previous page if available
+        // and to the ‘home’ folder if not possible to go back to the previous page
+        if (window.history.length !== 1) {
+            // browsers like chrome stores their homepage as first item
+            window.history.back();
+        } else {
+            window.location.href = ActionURL.getBaseURL(false);
+        }
     };
 
     onViewDetailsClick = (): void => {
-        this.setState(() => ({
-            showDetails: true,
-        }));
+        const { showDetails } = this.state;
+        this.setState({ showDetails: !showDetails });
```

Comment:
You'll want to use the state at the time of the callback.
```suggestion
        this.setState((state) => ({ showDetails: !state.showDetails }));
```